### PR TITLE
Redshift late-binding views don't play well with JDBC metadata description fix (#21215)

### PR DESCRIPTION
**Pull request by howonlee**
Pursuant to #21215.

There seems to be a bug materially identically in the stack trace to https://github.com/aws/amazon-redshift-jdbc-driver/issues/45 but not solved as of the 2.1.0.5 version that we encounter. To get around it, we just always use the fallback describe fields system for redshift case.

This doesn't seem to have perf impacts on local because it's redshift and web requests dominate either way - I need to go test for in-AWS-datacenter perf differences.

This is unfortunately a tad arbitrary as to the "weirdness if encountered arbitrarily" factor. However I tend to believe that it may have a better argument for that arbitrary if statement than one would expect - given our previous opinions hashed out on the idiosyncrasies of the first party redshift JDBC driver and the evident increased simplicity of the fallback field types

### CODE REVIEW

File: modules/drivers/redshift/test/metabase/driver/redshift_test.clj [hunk 1]
```diff
@@ -181,31 +181,47 @@
...
+            (is (= [{:name "numeric_col",   :database_type "numeric",         :base_type :type/Decimal}
+                    {:name "weird_varchar", :database_type "varchar", :base_type :type/Text}]
                    (map
                     (partial into {})
-                    (db/select [Field :name :database_type :base_type] :table_id table-id {:order-by [:name]}))))))))))
+                    (db/select [Field :name :database_type :base_type] :table_id table-id {:order-by [:name]}))))
+            (is (= [{:name "numeric_col_2",   :database_type "numeric",           :base_type :type/Decimal}
+                    {:name "weird_varchar_2", :database_type "varchar", :base_type :type/Text}]
+                   (map
+                     (partial into {})
```
Comment (dpsutton) [inline modules/drivers/redshift/test/metabase/driver/redshift_test.clj:223]: I think you are derecordizing here? If so we have `mt/derecordize`

**Comment by dpsutton** [comment]
I copied your tests but not your fix into my local version. Running
```clojure
redshift-test=> (redshift-types-test)
[logging output]
FAIL in (redshift-types-test) (redshift_test.clj:215)

:redshift Redshift specific types should be synced correctly
expected: [{:name "numeric_col", :database_type "numeric", :base_type :type/Decimal}
           {:name "weird_varchar", :database_type "varchar", :base_type :type/Text}]
  actual: ({:name "numeric_col", :database_type "numeric(10,2)", :base_type :type/Decimal}
           {:name "weird_varchar", :database_type "character varying(50)", :base_type :type/Text})
    diff: - [{:database_type "numeric"} {:database_type "varchar"}]
          + [{:database_type "numeric(10,2)"} {:database_type "character varying(50)"}]

FAIL in (redshift-types-test) (redshift_test.clj:220)

:redshift Redshift specific types should be synced correctly
expected: [{:name "numeric_col_2", :database_type "numeric", :base_type :type/Decimal}
           {:name "weird_varchar_2", :database_type "varchar", :base_type :type/Text}]
  actual: ({:name "numeric_col_2", :database_type "numeric(1,0)", :base_type :type/Decimal}
           {:name "weird_varchar_2", :database_type "text", :base_type :type/Text})
    diff: - [{:database_type "numeric"} {:database_type "varchar"}]
          + [{:database_type "numeric(1,0)"} {:database_type "text"}]
nil
```

All of these errors are related to "numeric" vs "numeric(1,0)". I don't see any that deal with the empty name which I think is the underlying error here?

(from the linked ticket):
> com.amazon.redshift.util.RedshiftException: ERROR: invalid input syntax for integer: ""
Which I think is coming from further processing after we have found empty names for columns.

```clojure
(defn- fallback-fields-metadata-from-select-query
  "In some rare cases `:column_name` is blank (eg. SQLite's views with group by) fallback to sniffing the type from a
  SELECT * query."
```
I'm seeing this is the fix but we don't have a failing test where `:column_name` is blank. Do you think we could add a test that exercises this?

**Comment by howonlee** [comment]
The `invalid input syntax for integer` actually comes from the redshift client lib. I will put in a bit exercising that bit.

File: modules/drivers/redshift/deps.edn [hunk 2]
```diff
@@ -5,4 +5,4 @@
  {"redshift" {:url "https://s3.amazonaws.com/redshift-maven-repository/release"}}
 
  :deps
- {com.amazon.redshift/redshift-jdbc42 {:mvn/version "2.1.0.3"}}}
+ {com.amazon.redshift/redshift-jdbc42 {:mvn/version "2.1.0.5"}}}
```
Comment (flamber) [inline modules/drivers/redshift/deps.edn:8]: Could we upgrade to latest 2.1.0.9 while we're at it? https://github.com/aws/amazon-redshift-jdbc-driver/blob/master/CHANGELOG.md

File: src/metabase/driver/sql_jdbc/sync/describe_table.clj [hunk 3]
```diff
@@ -124,14 +124,24 @@
...
         ;; VERY IMPORTANT! DO NOT REWRITE THIS TO BE LAZY! IT ONLY WORKS BECAUSE AS NORMAL-FIELDS GETS REDUCED,
         ;; HAS-FIELDS-WITHOUT-TYPE-INFO? WILL GET SET TO TRUE IF APPLICABLE AND THEN FALLBACK-FIELDS WILL RUN WHEN
         ;; IT'S TIME TO START EVALUATING THAT.
         (reduce
-         ((comp cat (m/distinct-by :name)) rf)
-         init
-         [jdbc-metadata fallback-metadata])))))
+          ((comp cat (m/distinct-by :name)) rf)
+          init
+          (if (= driver :redshift)
```
Comment (camsaul) [inline src/metabase/driver/sql_jdbc/sync/describe_table.clj:142]: Never use `=` to compare drivers, use `isa?` and `driver/hierarchy`. Maybe someone out there has created a driver based on our `:redshift` driver with some custom tweaks they need -- this stuff should apply to it

**Comment by noahmoss** [comment]
Closing this PR since I've merged a separate PR to resolve the issue

**Pull request closed without merging**